### PR TITLE
HDL-TH Integration support

### PR DESCRIPTION
**Pull request by kimpaller**
Adds the following features:

1. netbox: modify tags and add entry to journal
2. netbox: disable/enable board (device_status: Active, Offline, Fail) and describe action in journal
3. manager: verify checksum of written bootfiles via ssh

### CODE REVIEW

**Review by jpineda3 — COMMENTED** [review]
Do these changes require updating anything in jenkins pipeline configs?

File: tests/test_downloader.py
Comment (jpineda3) [inline tests/test_downloader.py:1]: Can the assertions be the following instead:

- hashes.txt is not empty
- files in outs in equal to hashes in file + 1

**Comment by kimpaller** [comment]
> Do these changes require updating anything in jenkins pipeline configs?

No, it should not affect existing pipeline definitions,

File: nebula/manager.py [hunk 1]
```diff
@@ -919,14 +919,24 @@ def shutdown_powerdown_board(self):
...
-    # p = os.path.join(p[0], "resources", "nebula-zed-fmcomms2.yaml")
-
-    # m = manager(configfilename=p)
-    # m.run_test()
-    pass
+        with open(hash_file, "r") as file:
+            for line in file:
+                fname, hash = line.strip().split(",")
+                # exclude some files
+                if fname in ["bootgen_sysfiles.tgz"]:
```
Comment (tagoylo) [inline nebula/manager.py:938]: Just curious why this file is excluded?

Comment (kimpaller) [reply nebula/manager.py:938]: For some reason, this file gets differnt hash wherever its place., and secondly, this file is not used in the boot sequence

File: nebula/tasks.py [hunk 2]
```diff
@@ -1402,8 +1422,154 @@ def check_board_booted(
...
+        yamlfilename=yamlfilename,
+        board_name=board_name,
+        load_config=load_config,
+    )
+    device = nebula.NetboxDevice(nb)
+    log.info(f"{board_name} status is {device.status()}")
+    if fail_if_inactive:
+        if not str(device.status()) == "Active":
+            raise Exception(f"{board_name} not Active")
+    return
```
Comment (tagoylo) [inline nebula/tasks.py:1562]: Aside from the log.info, will this also return the device status? Similar when using update-config?

File: nebula/tasks.py [hunk 3]
```diff
@@ -1402,8 +1422,154 @@ def check_board_booted(
...
+        ip=netbox_ip,
+        port=netbox_port,
+        token=netbox_token,
+        base_url=netbox_baseurl,
+        yamlfilename=yamlfilename,
+        board_name=board_name,
+        load_config=load_config,
+    )
+    device = nebula.NetboxDevice(nb)
+    log.info(f"{board_name} status is {device.status()}")
```
Comment (tagoylo) [inline nebula/tasks.py:1558]: Aside from log.info, will this also return the device status? Similar to update-config

File: nebula/tasks.py [hunk 4]
```diff
@@ -914,11 +915,30 @@ def update_boot_files_manager(
...
 
 
+@task(
+    help={
+        "folder": "Resource folder containing BOOT.BIN, kernel, device tree, and system_top.bit.\nOverrides other setting",
+        "yamlfilename": "Path to yaml config file. Default: /etc/default/nebula",
+        "board_name": "Name of DUT design (Ex: zynq-zc706-adv7511-fmcdaq2). Require for multi-device config files",
+    },
+)
+def verify_checksum_manager(
```
Comment (tagoylo) [inline nebula/tasks.py:925]: So running the verify checksum will be run on jenkins shared library, right after downloading the files?

**Review by tagoylo — COMMENTED** [review]
Added some questions

File: nebula/tasks.py (hunk 3 above)
Comment (kimpaller) [reply nebula/tasks.py:1558]: right now it isnt, but i could add a print() clause

File: nebula/tasks.py (hunk 4 above)
Comment (kimpaller) [reply nebula/tasks.py:925]: it should be within the UpdateBootFiles stage but after running nebula's update-boot-files (after succesful reboot with new boot files loaded) 

I will make a PR to JSL after this PR is merge:

https://github.com/sdgtt/jenkins-shared-library/blob/6a2711f092d8fd04ba8694e2abc9dc24eed01377/src/sdg/Gauntlet.groovy#L244

File: nebula/tasks.py (hunk 3 above)
Comment (kimpaller) [reply nebula/tasks.py:1558]: https://github.com/sdgtt/nebula/pull/122/commits/c31744c48017ec0a6447a4a6722928eee88c9bf1

File: nebula/manager.py [hunk 5]
```diff
@@ -919,14 +919,24 @@ def shutdown_powerdown_board(self):
             # force shutdown boards via pdu
             self.power.power_down_board()
 
+    def verify_checksum(self, folder):
+        log.info(f"Verifying bootfiles checksum for {self.board_name}")
+        # find hashes.txt on the folder
+        hash_file = None
+        for root, dirs, files in os.walk(folder):
+            if "hashes.txt" in files:
```
Comment (bia1708) [inline nebula/manager.py:927]: Should this file be manually created?

Comment (bia1708) [reply nebula/manager.py:927]: Nevermind, I just saw that the file is created in downloader.py

**Pull request merged**